client/src/components/visibility/Visibility.tsx

```diff
@@ -132,107 +173,151 @@ const VisibilityInput = ({
 
   const visibilities = computeVisibilities([namespaceVisibility]);
   const markInvalid = !visibility && isInvalid && isRequired;
-  const items = [
-    {
-      title: "Public",
-      value: "public",
-      icon: faGlobe,
-      hint: "Access without authentication",
-    },
-    {
-      title: "Internal",
-      value: "internal",
-      icon: faUserFriends,
-      hint: "Access only for authenticated users",
-    },
-    {
-      title: "Private",
-      value: "private",
-      icon: faLock,
-      hint: "Access only for the creator or contributors",
-    },
-  ];
-
-  const options = items.map((item) => {
-    const disabledByNamespace = visibilities.disabled.includes(item.value);
-    const isDisabled = disabled || disabledByNamespace;
 
+  const disableByNamespaceOptions = {
+    public: "",
+    private: `Public and Internal options are not available due to ${
+      isForked ? "forked project or " : ""
+    }group namespace restrictions. `,
+    internal: `Public is not available due to ${
```

Comment:
```suggestion
    internal: `Public not available due to ${
```
same pattern as the previous message